jupyter_server_fileid/manager.py

```diff
@@ -443,6 +444,12 @@ def get_path(self, id, sync=True):
         if ino != stat_info.ino or not self._check_timestamps(stat_info):
             return None
 
+        # if path is not relative to `self.root_dir`, return None.
+        if os.path.commonpath([self.root_dir, path]) != self.root_dir:
+            return None
+
+        # finally, convert the path to a relative one.
+        path = os.path.relpath(path, self.root_dir)
```

Comment:
Don't you need to [expand the user](https://github.com/jupyterlab/jupyterlab/blob/2e6921af6150661a76eeeca0f98220d6e1f1d5ed/jupyterlab/labapp.py#L685) (to convert `~` into e.g. `/home/david`)?

Reply:
Good question! The paths stored in the Files table are completely normalized and absolute, meaning they should never contain `~/`, `./`, or `../`. If you are noticing that the returned paths contain any of those substrings, please file a bug report.

Reply:
I think @davidbrochart is referring to the value in `self.root_dir`...
```bash
>>> import os
>>> os.path.relpath("/Users/kbates/notebooks/experiments", "~/notebooks")
'../../../../../../../../../notebooks/experiments'
>>> os.path.relpath("/Users/kbates/notebooks", os.path.expanduser("~/notebooks"))
'.'
```
However, the _currently_ configurable `root_dir` [is validated to ensure its value is absolute](https://github.com/jupyter-server/jupyter_server_fileid/blob/105415d82aa175553b3af9b498acf8af56da708a/jupyter_server_fileid/manager.py#L100).  I use _currently_ because this attribute should probably derive from the associated `ContentsManager` per #24 and not be user-configurable.  In that case, the FileID service will likely want to perform the expansion.